adal/src/main/java/com/microsoft/aad/adal/AuthenticationException.java

```diff
@@ -33,6 +41,12 @@
 
     private ADALError mCode;
 
+    private Map<String, String> mResponseBody;
+
+    private int mServiceStatusCode = -1;
+
+    private Map<String, List<String>> mHttpResponseHeaders;
```

Comment:
Hi @iambmelt I am wondering if it will be better if we combine the mResponseBody and mHttpResponseHeaders into a single Map<String, List<String>> map? 
Pros: it will make the class more extendable 
Cons: body and headers will be stored together

How do you think?

Reply:
I think I'm _personally_ preferential to keeping them separate. Given that headers and the response body are two distinct and separate parts of the HTTP response, it seems logical to me that they would be exposed through separate API calls on this object.